xpallets/gateway/records/src/lib.rs

```diff
@@ -173,9 +117,70 @@ decl_module! {
             Ok(())
         }
     }
+
+    #[pallet::event]
+    #[pallet::metadata(T::AccountId = "AccountId", BalanceOf<T> = "Balance", WithdrawalRecordOf<T> = "WithdrawalRecord")]
+    #[pallet::generate_deposit(pub(crate) fn deposit_event)]
+    pub enum Event<T: Config> {
+        /// An account deposited some asset. [who, asset_id, amount]
+        Deposited(T::AccountId, AssetId, BalanceOf<T>),
+        /// A withdrawal application was created. [withdrawal_id, record_info]
+        WithdrawalCreated(WithdrawalRecordId, WithdrawalRecordOf<T>),
+        /// A withdrawal proposal was processed. [withdrawal_id]
+        WithdrawalProcessed(WithdrawalRecordId),
+        /// A withdrawal proposal was recovered. [withdrawal_id]
+        WithdrawalRecovered(WithdrawalRecordId),
+        /// A withdrawal proposal was canceled. [withdrawal_id, withdrawal_state]
+        WithdrawalCanceled(WithdrawalRecordId, WithdrawalState),
+        /// A withdrawal proposal was finished successfully. [withdrawal_id, withdrawal_state]
+        WithdrawalFinished(WithdrawalRecordId, WithdrawalState),
+    }
+
+    #[pallet::error]
+    pub enum Error<T> {
+        /// Id not in withdrawal records
+        NotExisted,
+        /// WithdrawalRecord state not `Applying`
+        NotApplyingState,
+        /// WithdrawalRecord state not `Processing`
+        NotProcessingState,
+        /// The applicant is not this account
+        InvalidAccount,
+        /// State only allow `RootFinish` and `RootCancel`
+        InvalidState,
+        /// Meet unexpected chain
+        UnexpectedChain,
+    }
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(crate) trait Store)]
+    pub struct Pallet<T>(PhantomData<T>);
```

Comment:
```suggestion
   #[pallet::config]
   pub trait Config {
       ....
   }
   
    #[pallet::pallet]
    #[pallet::generate_store(pub(crate) trait Store)]
    pub struct Pallet<T>(PhantomData<T>);
```

Put `Pallet` right below `Config`.